frontend/src/pages/create_campaign/Roles/index.js

```diff
@@ -0,0 +1,297 @@
+    const newID = uuidv4();
+    if (newRoleName) {
+      setRoles([
+        ...roles,
+        {
+          id: newID,
+          title: newRoleName,
+          quantity: newRoleQty,
+        },
+      ]);
+      setNewRoleName("");
+      setNewRoleQty(1);
+      if (roleSelected === -1) {
+        setRoleSelected(newID);
+      }
+    } else {
+      alert("Role name is required!");
+    }
+  };
+
+  return (
+    <div>
+      {value.tab === index && (
+        <ContainerDiv>
+          <RolesDiv>
+            <SectionTitle>Roles</SectionTitle>
+            <RolesDisplay>
+              <RolesListContainer>
+                {roles.map((r) => (
+                  <>
+                    <RoleListItem selected={r.id === roleSelected}>
+                      <RoleListItemButton onClick={() => setRoleSelected(r.id)}>
+                        <RoleQuantity>{r.quantity}</RoleQuantity>
+                        <ListItemText>{r.title}</ListItemText>
+                        <ListItemIcon>
+                          <IconButton>
+                            <EditIcon fontSize="small" />
+                          </IconButton>
+                        </ListItemIcon>
+                        <ListItemIcon>
+                          <IconButton
+                            value={r.id}
+                            onClick={(e) => {
+                              onRoleDelete(e);
+                            }}
+                          >
+                            <ClearIcon />
+                          </IconButton>
+                        </ListItemIcon>
+                      </RoleListItemButton>
+                    </RoleListItem>
+                    <Divider />
+                  </>
+                ))}
+              </RolesListContainer>
+              <CreateRoleFormControl>
+                <CreateRoleFormGroup row>
+                  <CreateRoleQuantity
+                    type="number"
+                    value={newRoleQty}
+                    onChange={(e) => setNewRoleQty(e.target.value)}
+                    min={1}
+                  />
+                  <CreateRoleName
+                    type="text"
+                    value={newRoleName}
+                    onChange={(e) => setNewRoleName(e.currentTarget.value)}
+                  />
+                  <IconButton onClick={addRole}>
+                    <AddIcon />
+                  </IconButton>
+                </CreateRoleFormGroup>
+              </CreateRoleFormControl>
+            </RolesDisplay>
+          </RolesDiv>
+          <QuestionsDiv>
+            <SectionTitle>Questions</SectionTitle>
+            <QuestionsHeader>
+              <AddQuestionButton onClick={addQuestion} variant="outlined">
+                Add Question
+                <AddIcon fontSize="small" />
+              </AddQuestionButton>
+              <SelectFromExistingButton
+                variant="outlined"
+                aria-controls={open ? "basic-menu" : undefined}
```

Comment:
ah so i see you used `Bool(anchorEl)` - when you do ternary operators you can actually just put `anchorEl ? "basic-menu" : undefined` because it automatically gets converted to boolean

On another note though, it's not that good practice to have an undefined prop value, so I would rather have

```
{anchorEl ? (
<SelectFromExistingButton variant="outlined" aria-haspopup onClick={...}`
) : (
<SelectFromExistingButton ... (and put in the props here)
)
}

Reply:
This structure was literally just copied from MUI's sample code D: - can change it to whatever :)